avocado/core/status/repo.py

```diff
@@ -113,3 +113,9 @@ def get_task_status(self, task_id):
     @property
     def status_journal_summary(self):
         return self._status_journal_summary
+
+    def get_result_set_for_tasks(self, task_ids):
+        """Returns a set of results for the given task."""
+        results = [key for key, value in self._by_result.items()
+                   if any([True for task_id in task_ids if task_id in value])]
```

Comment:
I am not a huge fan of list comprehension with nested loops. For the nested loop I would use the inner function something  like `is_in_tasks(value)` for better readability, but it is up to you.

Reply:
+1 here

Reply:
Sure, I can do that.